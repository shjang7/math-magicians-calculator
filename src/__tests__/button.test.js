import React from 'react';
import { render } from '@testing-library/react';
import Button from '../components/button';

const orange = '#f60';

describe('<Button />', () => {
  const defaultButton = <Button buttonName="+" value="+" />;
  const wideButton = <Button buttonName="0" value="0" color="#aaa" wide />;

  it('matches snapshot', () => {
    const { container } = render(wideButton);
    expect(container).toMatchSnapshot();
  });

  it('shows the props correctly for button name', () => {
    const { getByText } = render(defaultButton);
    getByText('+');
  });

  it('shows the color correctly for button color', () => {
    const { queryByText } = render(defaultButton);
    queryByText(orange);
  });

  it('shows the width correctly for normal button', () => {
    const { queryByText } = render(defaultButton);
    queryByText('width: 25%');
  });

  it('shows the width correctly for wide button', () => {
    const { queryByText, getByText } = render(wideButton);
    getByText('0');
    queryByText('width: 50%');
  });
});
