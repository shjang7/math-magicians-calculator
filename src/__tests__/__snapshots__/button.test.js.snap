// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<div>
  <button
    class="button"
    style="background-color: rgb(170, 170, 170); width: 50%;"
    value="0"
  >
    0
  </button>
</div>
`;
